ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -1173,10 +1185,12 @@ public Node visitWhenClause(final SqlBaseParser.WhenClauseContext context) {
 
     @Override
     public Node visitFunctionCall(final SqlBaseParser.FunctionCallContext context) {
+      final List<Expression> expressionList = visit(context.expression(), Expression.class);
+      expressionList.addAll(visit(context.lambdaFunction(), Expression.class));
       return new FunctionCall(
```

Comment:
I'm going to update the g4 so that all the lambda functions have to be after all the other function arguments
```
    | identifier'(' (expression (',' expression)* (',' lambdaFunction)*)? ')'             #functionCall
```
So then this assumption can be made soundly

Reply:
Sounds good to me.